src/Microsoft.Azure.WebJobs.Host/Tables/TableAttributeBindingProvider.cs

```diff
@@ -68,32 +62,156 @@ private TableAttributeBindingProvider(INameResolver nameResolver, IStorageAccoun
 
             return new ResolvedTableAttribute(attrResolved, client);
         }
-        
+
         public static IBindingProvider Build(INameResolver nameResolver, IConverterManager converterManager, IStorageAccountProvider accountProvider, IExtensionRegistry extensions)
         {
             var original = new TableAttributeBindingProvider(nameResolver, accountProvider, extensions);
 
             converterManager.AddConverter<JObject, ITableEntity, TableAttribute>(original.JObjectToTableEntityConverterFunc);
+            converterManager.AddConverterBuilder<object, ITableEntity, TableAttribute>(original.BuildITEConverter);
+
+            // IStorageTable --> IQueryable<ITableEntity>
+            converterManager.AddConverterBuilder<IStorageTable, TableQueryableOpenType, TableAttribute>(
+                original.QueryableConverter);
 
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
-            var bindAsyncCollector = bindingFactory.BindToAsyncCollector<TableAttribute, ITableEntity>(original.BuildFromTableAttribute, null, original.CollectAttributeInfo);
 
-            var bindToJobject = bindingFactory.BindToExactAsyncType<TableAttribute, JObject>(original.BuildJObject, null, original.CollectAttributeInfo);
-            var bindToJArray = bindingFactory.BindToExactAsyncType<TableAttribute, JArray>(original.BuildJArray, null, original.CollectAttributeInfo);
+            var bindToExactCloudTable = bindingFactory.BindToGeneralAsyncType<TableAttribute, CloudTable>(
```

Comment:
The func callback is async. This could benefit from a better name. 

---
In reply to: [95847425](https://github.com/Azure/azure-webjobs-sdk/pull/967#discussion_r95847425) [](ancestors = 95847425)